### PR TITLE
feat(wallets): Add sanchonet wallet option

### PR DESCRIPTION
**Pull request by michalszorad**
Resolves 
- [YOMO-1024](https://emurgo.atlassian.net/browse/YOMO-1024)
- [YOMO-1025](https://emurgo.atlassian.net/browse/YOMO-1025)

Faucet: https://sancho.network/faucet

Screenshot 2023-12-11 at 19 40 39

Screenshot 2023-12-11 at 19 42 58

[YOMO-1024]: https://emurgo.atlassian.net/browse/YOMO-1024?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[YOMO-1025]: https://emurgo.atlassian.net/browse/YOMO-1025?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/wallet-mobile/src/yoroi-wallets/cardano/constants/sanchonet/constants.ts [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+} as const
+
+const CARDANO_BASE_CONFIG = [BYRON_BASE_CONFIG, SHELLEY_BASE_CONFIG]
+export const NETWORK_CONFIG = {
+  BACKEND,
+  BASE_CONFIG: CARDANO_BASE_CONFIG,
+  CHAIN_NETWORK_ID: CHAIN_NETWORK_ID.toString(),
+  COIN_TYPE,
+  ENABLED: true,
+  EXPLORER_URL_FOR_ADDRESS: (address: string) => `https://preprod.cardanoscan.io/address/${address}`,
```
Comment (michalszorad) [inline apps/wallet-mobile/src/yoroi-wallets/cardano/constants/sanchonet/constants.ts:59]: No sanchonet explorers yet

File: apps/wallet-mobile/src/features/Staking/Governance/common/LearnMoreLink/LearnMoreLink.tsx [hunk 2]
```diff
@@ -5,8 +5,7 @@ import {Text} from '../../../../../components'
 import {COLORS} from '../../../../../theme'
 import {useStrings} from '../strings'
 
-// TODO: replace with real link
-const LEARN_MORE_LINK = 'https://google.com'
+const LEARN_MORE_LINK = ''
```
Comment (michalszorad) [inline apps/wallet-mobile/src/features/Staking/Governance/common/LearnMoreLink/LearnMoreLink.tsx:8]: Links will be hidden until we have them ready [YOMO-1025](https://emurgo.atlassian.net/browse/YOMO-1025)

File: apps/wallet-mobile/src/features/Staking/Governance/useCases/EnterDrepIdModal/EnterDrepIdModal.tsx [hunk 3]
```diff
@@ -11,6 +11,8 @@ type Props = {
   onSubmit?: (drepId: string) => void
 }
 
+const FIND_DREPS_LINK = ''
```
Comment (michalszorad) [inline apps/wallet-mobile/src/features/Staking/Governance/useCases/EnterDrepIdModal/EnterDrepIdModal.tsx:14]: Links will be hidden until we have them ready [YOMO-1025](https://emurgo.atlassian.net/browse/YOMO-1025)

File: apps/wallet-mobile/src/yoroi-wallets/cardano/constants/sanchonet/constants.ts [hunk 4]
```diff
@@ -0,0 +1,108 @@
+import {Balance} from '@yoroi/types'
+
+import {DefaultAsset} from '../../../types'
+import {COIN_TYPE, COINS_PER_UTXO_WORD, KEY_DEPOSIT, LINEAR_FEE, MINIMUM_UTXO_VAL, POOL_DEPOSIT} from '../common'
+
+export * from '../common'
+
+export const NETWORK_ID = 4_50
```
Comment (stackchain) [inline apps/wallet-mobile/src/yoroi-wallets/cardano/constants/sanchonet/constants.ts]: `450` is fine, plenty of `isMainnet` checks that are considering `300` only will require update.

Comment (michalszorad) [reply apps/wallet-mobile/src/yoroi-wallets/cardano/constants/sanchonet/constants.ts]: I've added `isMainnetNetworkId` helper and used it across `wallet-mobile`

File: apps/wallet-mobile/src/features/Staking/Governance/useCases/Home/HomeScreen.tsx [hunk 5]
```diff
@@ -230,6 +232,12 @@ const NeverParticipatedInGovernanceVariant = () => {
   )
 }
 
+const useRerenderOnFocus = () => {
```
Comment (jorbuedo) [inline apps/wallet-mobile/src/features/Staking/Governance/useCases/Home/HomeScreen.tsx]: This looks bad, I guess it's solving some bug that appears without it? Is there no other way?

**Review by jorbuedo — APPROVED** [review]

File: apps/wallet-mobile/src/features/Staking/Governance/useCases/Home/HomeScreen.tsx (hunk 5 above)
Comment (michalszorad) [reply apps/wallet-mobile/src/features/Staking/Governance/useCases/Home/HomeScreen.tsx]: I'll check and send an update

Comment (michalszorad) [reply apps/wallet-mobile/src/features/Staking/Governance/useCases/Home/HomeScreen.tsx]: That's now resolved. There was an issue when invalidating some queries

**Pull request merged**